src/retrie/trie.py

```diff
@@ -35,20 +52,81 @@ class Trie:
 
     __slots__ = "data"
 
-    def __init__(self):
+    def __init__(
+        self, *word  # type: Text
+    ):
         """Initialize data dictionary."""
         self.data = {}  # type: data_type
+        self.add(*word)
+
+    def __eq__(
+        self, other  # type: Any
+    ):  # type: (...) -> bool
+        """Compare two Trie objects."""
+        if not isinstance(other, Trie):
+            return False
+        return self.data == other.data
+
+    def __add__(
+        self, other  # type: "Trie"
+    ):  # type: (...) -> "Trie"
+        """Merge two Trie objects."""
+        new_trie = Trie()
+        new_trie += self
+        new_trie += other
+        return new_trie
+
+    def __iadd__(
+        self,
+        other,  # type: "Trie"
+    ):  # type: (...) -> "Trie"
+        """Merge another Trie object into the current Trie."""
+        if self.__class__ != other.__class__:
+            raise TypeError(
+                "Unsupported operand type(s) for +=: '{0}' and '{1}'".format(
+                    type(self), type(other)
+                )
+            )
+        self._merge(other)
+        return self
+
+    def _merge(
+        self,
+        other,  # type: "Trie"
+    ):  # type: (...) -> None
+        """Merge two Trie data."""
+        for key, value in other.data.items():
+            if key in self.data:
+                self._merge_subtrie(self.data[key], value)
+            else:
+                self.data[key] = value
+
+    def _merge_subtrie(
+        self,
+        current_subtrie,  # type: data_type
+        other_subtrie,  # type: data_type
+    ):  # type: (...) -> None
+        """Recursively merge subtrie data."""
+        for key, value in other_subtrie.items():
+            if key in current_subtrie:
+                self._merge_subtrie(current_subtrie[key], value)
+            else:
+                current_subtrie[key] = value
 
     def add(
         self, *word  # type: Text
-    ):
+    ):  # type: (...) -> "Trie"
         """Add one or more words to the current Trie."""
+        if len(word) == 1 and not isinstance(word[0], str):
+            word = word[0]
```

Comment:
I would like to avoid this behaviour change: if you have a list/set of strings completely in memory already, you can already achieve the semantic without this if-statement by unpacking the iterable with asterisk: `Trie(*["abc", "foo"])` or `Trie().add(*["abc", "foo"])`. Can you remove this if-statement and amend the examples?